cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java

```diff
@@ -46,16 +47,12 @@
  */
 public class CLIMain {
 
-  private final CLIConfig cliConfig;
-  private final HelpCommand helpCommand;
   private final CLI cli;
 
   private final Iterable<Command> commands;
```

Comment:
+1 for removing ``cliConfig`` and ``helpCommand`` as instance variables. Can we also make ``commands`` local to the constructor? I see that the ``getCommands()`` method is only used in the constructor, even though its public, unless I'm missing something.

Also, a cycle - ``helpCommand`` needs ``commands`` to be constructed, while ``commands`` contains ``helpCommands``. Wonder if that would cause any issues. 

Maybe unrelated to this PR, and have no way to get around, but worth thinking about.